### PR TITLE
Use namespaced modules for Vuex store

### LINKED ISSUE #205: [Feature] Break down megastore into smaller modules
## Problem

Currently we have a huge megastore that stores everything. This is not optimal because
- there is a possibility of name collisions in state (actions and mutations are safer due to use of constants)
- all information about the hierarchy of modules is lost when they are flattened and merged into one

Moreover modules are merged with `Object.assign` without a blank target, which alters the first argument to receive all subsequent keys.

## Description

Putting the contents of `store-modules/` inside `store/` can automatically configure Nuxt to use namespaced modules. The need for manually merging all stores with `Object.assign` is also eliminated.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #205 by @dhruvkb 

## Description

This PR refactors one mega-store that we currently have into namespaced modules inside the `store` directory.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [ ] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by dhruvkb** [comment]
Nearly a hundred files changed! That'll be an intense review (although all checks passed so that's a good sign).

**Comment by obulat** [comment]
> Nearly a hundred files changed! That'll be an intense review (although all checks passed so that's a good sign).

I guess I should exchange this intense review with the review of https://github.com/WordPress/openverse-api/pull/194 :)

With all store modules becoming namespaced, all calls of `commit`, `dispatch` and `mapState` had to be changed to add the namespace to them, that is one of the reasons for so many changes.
Another one is it has to be all or nothing change (or I don't know how to have both with the current structure).
The checks pass because I commented out some of the tests that fail because I haven't yet figured out how to mock namespaced actions

**Comment by zackkrida** [comment]
@obulat I haven't read this yet but it appears to talk about mocking namespaced actions https://al-un.github.io/learn-nuxt-ts/06.test.html#mocking-vuex-store I'll check later when I review this. 

So much work! That's why I didn't do this as part of the original Nuxt migration 😅

File: src/components/AudioResultsList.vue [hunk 1]
```diff
@@ -80,23 +87,15 @@ export default {
...
-    },
-    errorMessage() {
-      return this.$store.state.errorMessage
+      return this.currentPage >= this.$store.state.search.pageCount.audios
     },
   },
   methods: {
+    ...mapActions({
+      fetchMedia: `${SEARCH}/${FETCH_MEDIA}`,
+    }),
```
Comment (dhruvkb) [inline src/components/AudioResultsList.vue:98]: For binding namespaced actions, provide the name of the module as the first argument for the `mapActions` call.

```suggestion
    ...mapActions(SEARCH, [
      FETCH_MEDIA,
    ]),
```

Comment (zackkrida) [reply src/components/AudioResultsList.vue:98]: Ooh, that's a great suggestion @dhruvkb I didn't know you could do that. Bye string concatenation!

Comment (dhruvkb) [reply src/components/AudioResultsList.vue:98]: In that case we can do
```js
...mapActions(SEARCH, {
  fetchMedia: FETCH_MEDIA
}),
```

**Comment by obulat** [comment]
Closing this PR as it was replaced by #244 and #242.

**Pull request closed without merging**